pkg/bot/mattermost_fmt.go

```diff
@@ -0,0 +1,97 @@
+package bot
+
+import (
+	"fmt"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+
+	"github.com/kubeshop/botkube/pkg/events"
+	format2 "github.com/kubeshop/botkube/pkg/format"
+)
+
+func (b *Mattermost) longNotification(event events.Event) []*model.SlackAttachmentField {
+	fields := []*model.SlackAttachmentField{
+		{
+			Title: "Kind",
+			Value: event.Kind,
+			Short: true,
+		},
+		{
+			Title: "Name",
+			Value: event.Name,
+			Short: true,
+		},
+	}
+
+	if event.Namespace != "" {
+		fields = append(fields, &model.SlackAttachmentField{
+			Title: "Namespace",
+			Value: event.Namespace,
+			Short: true,
+		})
+	}
+
+	if event.Reason != "" {
+		fields = append(fields, &model.SlackAttachmentField{
+			Title: "Reason",
+			Value: event.Reason,
+			Short: true,
+		})
+	}
+
+	if len(event.Messages) > 0 {
+		message := ""
+		for _, m := range event.Messages {
+			message += fmt.Sprintf("%s\n", m)
+		}
+		fields = append(fields, &model.SlackAttachmentField{
+			Title: "Message",
+			Value: message,
+		})
+	}
+
+	if event.Action != "" {
+		fields = append(fields, &model.SlackAttachmentField{
+			Title: "Action",
+			Value: event.Action,
+		})
+	}
+
+	if len(event.Recommendations) > 0 {
+		rec := ""
+		for _, r := range event.Recommendations {
+			rec += fmt.Sprintf("%s\n", r)
+		}
+		fields = append(fields, &model.SlackAttachmentField{
+			Title: "Recommendations",
+			Value: rec,
+		})
+	}
+
+	if len(event.Warnings) > 0 {
+		warn := ""
+		for _, w := range event.Warnings {
+			warn += fmt.Sprintf("%s\n", w)
+		}
+
+		fields = append(fields, &model.SlackAttachmentField{
+			Title: "Warnings",
+			Value: warn,
+		})
+	}
```

Comment:
all of this can be replaced with:

```go
	join := func(in []string) string {
		return strings.Join(in, "\n")
	}

	fields = b.appendIfNotEmpty(fields, event.Namespace, "Namespace", true)
	fields = b.appendIfNotEmpty(fields, event.Reason, "Reason", true)
	fields = b.appendIfNotEmpty(fields, event.Action, "Action", true)

	fields = b.appendIfNotEmpty(fields, join(event.Messages), "Message", false)
	fields = b.appendIfNotEmpty(fields, join(event.Recommendations), "Recommendations", false)
	fields = b.appendIfNotEmpty(fields, join(event.Warnings), "Warnings", false)
```

and the func:
```go
func (b *Mattermost) appendIfNotEmpty(fields []*model.SlackAttachmentField, in string, title string, short model.SlackCompatibleBool) []*model.SlackAttachmentField {
	if in == "" {
		return fields
	}
	return append(fields, &model.SlackAttachmentField{
		Title: title,
		Value: in,
		Short: short,
	})
}
```

the same suggestion for other Bot's "formatters"

Reply:
Yep, I didn't touch the bot formatters, just extracted that to a separate file. Thanks, will apply the changes anyway!